### PR TITLE
[QS-9-10] Reject attestations that fail ancestry condition

### PR DESCRIPTION
**Pull request by cemozerr**
## PR Description
Reject attestations that fail ancestry condition.

## Documentation

- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

### CODE REVIEW

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/topics/validation/AttestationValidator.java [hunk 1]
```diff
@@ -191,6 +191,21 @@ private InternalValidationResult singleAttestationChecks(final Attestation attes
...
+              // aggregate.data.beacon_block_root
+              if (!forkChoiceUtilWrapper
+                  .get_ancestor(
+                      recentChainData.getForkChoiceStrategy().orElseThrow(),
+                      data.getBeacon_block_root(),
+                      compute_start_slot_at_epoch(recentChainData.getFinalizedEpoch()))
+                  .map(
+                      ancestorOfLMDVote ->
+                          ancestorOfLMDVote.equals(
+                              recentChainData.getFinalizedCheckpoint().orElseThrow().getRoot()))
```
Comment (Nashatyrev) [inline networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/topics/validation/AttestationValidator.java]: BTW can't it be the concurrency case when `recentChainData` updates its finalized checkpoint between calls `recentChainData.getFinalizedEpoch()` and `recentChainData.getFinalizedCheckpoint()`? 
May be get `recentChainData.getFinalizedCheckpoint()` once into local var to be certainly safe on this?

**Review by Nashatyrev — APPROVED** [review]
LGTM, just a comment on possible concurrency case

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/topics/validation/AttestationValidator.java (hunk 1 above)
Comment (cemozerr) [reply networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/topics/validation/AttestationValidator.java]: Good point. Will make the change.

Comment (cemozerr) [reply networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/topics/validation/AttestationValidator.java]: Done.

**Pull request merged**